Composer/packages/client/src/components/CreationFlow/v2/CreateBot.tsx

```diff
@@ -200,27 +192,35 @@ export function CreateBotV2(props: CreateBotProps) {
   };
 
   const getTemplate = (): BotTemplate | undefined => {
-    const currentTemplate = templates.find((t) => {
-      return t.id === currentTemplateId;
+    const currentTemplate = displayedTemplates.find((t) => {
+      return t?.id === currentTemplateId;
     });
     return currentTemplate;
   };
 
   useEffect(() => {
-    if (templates.length > 1) {
-      const emptyBotTemplate = find(templates, ['id', defaultTemplateId]);
-      if (emptyBotTemplate) {
-        setCurrentTemplateId(emptyBotTemplate.id);
-        setEmptyBotKey(emptyBotTemplate.id);
+    const itemKey = selectedProgLang?.props?.itemKey;
+
+    if (itemKey) {
+      if (itemKey === csharpFeedKey) {
+        const newTemplates = templates.filter((template) => {
+          return template.dotnetSupport;
+        });
+        setDisplayedTemplates(newTemplates);
+      } else if (itemKey === nodeFeedKey) {
+        const newTemplates = templates.filter((template) => {
+          return template.nodeSupport;
+        });
+        setDisplayedTemplates(newTemplates);
       }
```

Comment:
If `csharpFeedKey` and `nodeFeedKey` will never themselves be null, I think you can get rid of the outer `if`.

Reply:
Sounds good, I have removed the outer if, the logic should ensure that it is never null